python/gsw/main.py

```diff
@@ -0,0 +1,370 @@
+        self.run_email_thread = True
+        self.check_email_thread.start()
+
+    def process_downlink_packet(self, data):
+        '''
+        Converts the email attachment data into a 
+        JSON object. This is because elasticsearch
+        only takes in JSON data. I will add to this
+        and organize the attachment contents more
+        once downlink producer is finished. 
+        '''
+        data["time"]=str(datetime.now().isoformat())
+        return json.loads(data)
+
+    def check_for_email(self):
+        '''
+        Reads the most recent unread email from the Iridium email account. 
+
+        If it is a downlink, the function returns a statefield report. If it is an uplink 
+        confirmation, the function records that we recieved an uplink confirmation and returns
+        None. If there is neither a downlink or an uplink, the function returns None.
+
+        If either a downlink or an uplink is recieved, the function will update the most recent MOMSN 
+        and MTMSN numbers and check whether or not radioSession can send uplinks.
+        '''
+        #look for all new emails from iridium
+        _, data = self.mail.search(None, '(FROM "sbdservice@sbd.iridium.com")', '(UNSEEN)')
+        mail_ids = data[0]
+        id_list = mail_ids.split()
+
+        for num in id_list:
+            #.fetch() fetches the mail for given id where 'RFC822' is an Internet 
+            # Message Access Protocol.
+            _, data = self.mail.fetch(num,'(RFC822)')
+
+            #go through each component of data
+            for response_part in data:
+                if isinstance(response_part, tuple):
+                    # converts message from byte literal to string removing b''
+                    msg = email.message_from_string(response_part[1].decode('utf-8'))
+                    email_subject = msg['subject']
+
+                    #handles uplink confirmations
+                    if email_subject.find("SBD Mobile Terminated Message Queued for Unit: ")==0:
+                        # Get imei number of the radio that recieved the uplink
+                        self.imei=int(email_subject[47:])
+
+                        for part in msg.walk():
+                                            
+                            if part.get_content_maintype() == 'multipart':
+                                continue
+
+                            if part.get('Content-Disposition') is None:
+                                continue
+
+                            #get the body text of the email and look for the MOMSN/MTMSN number
+                            if part.get_content_type() == "text/plain":
+                                email_body = part.get_payload(decode=True).decode('utf8')
+                                for line in email_body.splitlines():
+                                    if line.find("MTMSN")!=-1:
+                                        self.mtmsn=int(line[line.find("MTMSN")+9:line.find("MTMSN")+11])
```

Comment:
You use the exact same code below in processing downlinks. Consider breaking it out to a function, so that we minimize code duplication. (Breaking it out into a function will also make things easier to read)

Reply:
Nevermind, the code isn't equal! I misread.